apps/api/src/services/boost.service.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  type BoostAgentResult,
+  BoostRepository,
+} from "@/database/repositories/boost.repository.js";
+import { serviceLogger } from "@/lib/logger.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import type { CompetitionService } from "@/services/competition.service.js";
+import type { UserService } from "@/services/user.service.js";
+
+/**
+ * Parameters for boosting an agent
+ */
+export interface BoostAgentParams {
+  userId: string;
+  competitionId: string;
+  agentId: string;
+  amount: bigint;
+  idemKey: Buffer;
+}
+
+/**
+ * Boost Service
+ * Manages boost operations including validation and business logic
+ */
+export class BoostService {
+  private boostRepository: BoostRepository;
+  private competitionService: CompetitionService;
+  private userService: UserService;
+
+  constructor(
+    boostRepository: BoostRepository,
+    competitionService: CompetitionService,
+    userService: UserService,
+  ) {
+    this.boostRepository = boostRepository;
+    this.competitionService = competitionService;
+    this.userService = userService;
+  }
+
+  /**
+   * Get user boost balance for a competition
+   * @param userId The user ID
+   * @param competitionId The competition ID
+   * @returns The user's boost balance
+   */
+  async getUserBoostBalance(
+    userId: string,
+    competitionId: string,
+  ): Promise<bigint> {
+    try {
+      // Get the user to validate they exist
+      const user = await this.userService.getUser(userId);
+      if (!user) {
+        throw new ApiError(404, "User not found");
```

Comment:
Not saying we should change this now, but it's weird to have http errors in a class that has nothing to do with http. This class could be used in any context. We can revisit in the future.